GomokuServer/src/GomokuServer.Api/MatchingEngine/QuickPairingGameOptionsVariants.cs

```diff
@@ -0,0 +1,58 @@
+﻿namespace GomokuServer.Api.MatchingEngine;
+
+public static class QuickPairingGameOptionsVariants
+{
+	public readonly static List<GameOptions> Values;
+
+	public static bool IsValidGameOptions(GameOptions gameOptions) =>
+		Values.Contains(gameOptions);
+	
+	static QuickPairingGameOptionsVariants()
+	{
+		Values = [];
+		
+		foreach (var isAnonymous in new[] { true, false })
+		{
+			Values.AddRange([
+				new GameOptions(
+					13,
+					new TimeControlDto() { InitialTimeInSeconds = 60, IncrementPerMove = 0 },
+					isAnonymous),
+				new GameOptions(
+					13,
+					new TimeControlDto() { InitialTimeInSeconds = 60, IncrementPerMove = 1 },
+					isAnonymous),
+				new GameOptions(
+					13,
+					new TimeControlDto() { InitialTimeInSeconds = 60, IncrementPerMove = 2 },
+					isAnonymous),
+
+				new GameOptions(
+					13,
+					new TimeControlDto() { InitialTimeInSeconds = 120, IncrementPerMove = 1 },
+					isAnonymous),
+				new GameOptions(
+					13,
+					new TimeControlDto() { InitialTimeInSeconds = 300, IncrementPerMove = 0 },
+					isAnonymous),
+				new GameOptions(
+					17,
+					new TimeControlDto() { InitialTimeInSeconds = 420, IncrementPerMove = 0 },
+					isAnonymous),
+				
+				new GameOptions(
+					17,
+					new TimeControlDto() { InitialTimeInSeconds = 600, IncrementPerMove = 0 },
+					isAnonymous),
+				new GameOptions(
+					19,
+					new TimeControlDto() { InitialTimeInSeconds = 900, IncrementPerMove = 0 },
+					isAnonymous),
+				new GameOptions(
+					19,
+					new TimeControlDto() { InitialTimeInSeconds = 1800, IncrementPerMove = 0 },
+					isAnonymous)
+			]);
+		}
```

Comment:
Well, it does looks like we are not following DRY principles here, please consider the following: 

```
var gameOptionsData = new (int BoardSize, int InitialTime, int Increment)[]
{
    (13, 60, 0),
    (13, 60, 1),
    (13, 60, 2),
    (13, 120, 1),
    (13, 300, 0),
    (17, 420, 0),
    (17, 600, 0),
    (19, 900, 0),
    (19, 1800, 0)
};

foreach (var isAnonymous in new[] { true, false })
{
    foreach (var data in gameOptionsData)
    {
        Values.Add(new GameOptions(
            data.BoardSize,
            new TimeControlDto { InitialTimeInSeconds = data.InitialTime, IncrementPerMove = data.Increment },
            isAnonymous));
    }
}
```